src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -597,13 +629,50 @@ void AlgorithmPaintToolBox::copyMetaData(medAbstractData* output,
     medUtilities::setDerivedMetaData(output, input, "painted");
 }
 
+dtkSmartPointer<medAbstractData> AlgorithmPaintToolBox::applyThresholdToMask(medAbstractData* inpuMask)
+{
+    dtkSmartPointer <medAbstractProcess> thresholdProcess;
+    try
+    {
+        thresholdProcess = dtkAbstractProcessFactory::instance()->createSmartPointer ( "itkThresholdingProcess" );
+    }
+    catch( itk::ExceptionObject &err )
+    {
+        qDebug() << err.GetDescription();
+        return nullptr;
+    }
+
+    thresholdProcess->setInput(inpuMask);
+    double threshold = 0.;
+    double outputValue = 1.;
+    double upper = 1.; // boolean = true
+    thresholdProcess->setParameter(threshold, 0);
+    thresholdProcess->setParameter(outputValue, 1);
+    thresholdProcess->setParameter(upper, 2);
+    try
+    {
+        thresholdProcess->update();
+    }
+    catch( itk::ExceptionObject &err )
+    {
+        qDebug() << err.GetDescription();
+        return nullptr;
+    }
```

Comment:
ITK exceptions are already handled in```itkFiltersProcessBase::update()``` and will never go further down. Use the return value of ```update``` instead if you want to test the success of the function.